### PR TITLE
Multireduce Scheduler Tests

### PR DESCRIPTION
**Pull request by 0xtimmy**
Beginning of the scheduler tests for multireduce. This covers what I think are the obvious cases to look out for, I'll add some as I got to reflect whatever changes to the scheduler

### CODE REVIEW

File: test/test_schedule.py [hunk 1]
```diff
@@ -518,13 +528,28 @@ def test_reduce_shrink(self):
     out = x + y
     check_schedule(out, 2)  # TODO: this should be 1
 
+  # multireduce spec
+  @unittest.skip
+  def test_multireduce_shrink(self):
+    a = Tensor.empty(32, 32)
```
Comment (Qazalin) [inline test/test_schedule.py:534]: can you replace empty with randn so that we can test correctness?

**Review by Qazalin — COMMENTED** [review]
The best way to create scheduler specs is if you check the schedule kernel count with the current counts and then modify.

So you wouldn't skip tests. Once the fusion is complete we'd have a diff like this:
https://github.com/tinygrad/tinygrad/pull/4322/files

**Comment by Qazalin** [comment]
two more development experience points that'd help:
- Always seed the randn kernel
- Realize the Tensor.randn before check_schedule to cleanup DEBUG=4

**Comment by 0xtimmy** [comment]
bet

File: test/test_schedule.py [hunk 2]
```diff
@@ -602,10 +647,65 @@ def test_reduce_shrink_output(self):
     out1 = out0[0] + Tensor.empty(1, )
     check_schedule([r, out0, out1], 3)
 
+  # multireduce spec
+  @unittest.skip
+  def test_multireduce_fusion(self):
```
Comment (Qazalin) [inline test/test_schedule.py:652]: can you break this into multiple defs, ideally each having one check_schedule

File: test/test_schedule.py [hunk 3]
```diff
@@ -734,6 +834,19 @@ def test_reduce_simple_chase(self):
...
+  @unittest.skip
+  def test_multireduce_simple_chase(self):
+    a = Tensor.rand(4, 4, 4)
+    r = (a + (a.sum(0, keepdim=True) + 6)).sum(0) * 2
+    b = r.sum(0) + 8
+    c = r.sum(1) + 12
+    np_r = (a.numpy() + (a.numpy().sum(0) + 6)).sum(0) * 2
+    np.testing.assert_allclose(b.numpy(), np_r.sum(0) + 8)
+    np.testing.assert_allclose(c.numpy(), np_r.sum(1) + 12)
+    schedule = check_schedule([b, c], 3)
```
Comment (Qazalin) [inline test/test_schedule.py:847]: you can't reschedule b and c.
these should be `s = create_schedule()` then `run_schedule(s)`

**Pull request merged**